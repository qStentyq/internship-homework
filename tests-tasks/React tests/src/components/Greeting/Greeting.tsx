

export default function Greeting({name = ''}: {name?: string}) {
  return (
    <>
      {name ? 
        <h1>Hello {name}</h1> : 
        <h1>Hello Guest</h1>}
    </>
  )
}

/*<div className='guest_menu'>Guest 
          <input type="text" placeholder="Enter your name " value={inputVal} onChange={(e) => setInputVal(e.target.value)}/>
          <button onClick={() => setName(inputVal)}>Login</button>
      
        </div>*/
