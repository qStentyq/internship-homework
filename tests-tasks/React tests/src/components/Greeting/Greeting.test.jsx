import Greeting from './Greeting';
import { render, cleanup } from '@testing-library/react';
afterEach(cleanup);
const GREETINGS = {
  withName: "Hello John",
  withoutName: "Hello Guest"
};
describe('Greeting Component', () => {
  it.each([
    { name: "John", expectedText: GREETINGS.withName },
    { name: "", expectedText: GREETINGS.withoutName }
  ])('renders correctly with name: "$name"', ({ name, expectedText }) => {
    const { getByText } = render(<Greeting name={name} />);
    expect(getByText(new RegExp(expectedText, 'i'))).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(<Greeting name="John" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
