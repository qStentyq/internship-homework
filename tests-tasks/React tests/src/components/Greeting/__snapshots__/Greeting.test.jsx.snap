// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Greeting > renders correctly 1`] = `null`;

exports[`Greeting Component > matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Hello John
  </h1>
</DocumentFragment>
`;
